oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/config/annotation/web/configurers/OidcUserInfoEndpointConfigurer.java

```diff
@@ -82,13 +151,24 @@ void init(HttpSecurity httpSecurity) {
 				new AntPathRequestMatcher(userInfoEndpointUri, HttpMethod.GET.name()),
 				new AntPathRequestMatcher(userInfoEndpointUri, HttpMethod.POST.name()));
 
-		OidcUserInfoAuthenticationProvider oidcUserInfoAuthenticationProvider =
-				new OidcUserInfoAuthenticationProvider(
-						OAuth2ConfigurerUtils.getAuthorizationService(httpSecurity));
+		List<AuthenticationProvider> authenticationProviders = createDefaultAuthenticationProviders(httpSecurity);
+
 		if (this.userInfoMapper != null) {
-			oidcUserInfoAuthenticationProvider.setUserInfoMapper(this.userInfoMapper);
+			// @formatter:off
+			authenticationProviders.stream()
+					.filter(OidcUserInfoAuthenticationProvider.class::isInstance)
+					.map(OidcUserInfoAuthenticationProvider.class::cast)
+					.forEach(provider -> provider.setUserInfoMapper(this.userInfoMapper));
```

Comment:
This should be configured in `createDefaultAuthenticationProviders()`

Reply:
`createDefaultAuthenticationProviders` is currently static.

Do you prefer:
1. Static with param:
  ```java
  public static void createDefaultAuthenticationProviders(
    Function<OidcUserInfoAuthenticationContext, OidcUserInfo> userInfoMapper
  ) { ... }
  ```
2. Non static 
  ```java
  public void createDefaultAuthenticationProviders() {
    // [...]
    authenticationProvider.setUserInfoMapper(this.userInfoMapper);
    // [...]
  }
  ```

Reply:
Non-static. The same change was applied to `OAuth2AuthorizationEndpointConfigurer.createDefaultAuthenticationProviders()`